hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyPurgeRequest.java

```diff
@@ -74,14 +75,12 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
     try {
       SnapshotInfo fromSnapshotInfo = null;
       if (fromSnapshot != null) {
-        fromSnapshotInfo = ozoneManager.getMetadataManager()
-            .getSnapshotInfoTable().get(fromSnapshot);
+        fromSnapshotInfo = SnapshotUtils.getSnapshotInfo(ozoneManager, fromSnapshot);
```

Comment:
perhaps one more thing that we can do is to pass an additional flag to look from just cache where we expect the entry to be there. Other places where we are not sure, the flag could be true to allow read from disk is ok.

Reply:
I haven't changed `SnapshotUtils.getSnapshotInfo()`.

I just added a private function to `OmSnapshotManager` to use for `SnapshotCache` because it needs to know the snapshot dir to open a RocksDB instance.